test/jdk/java/nio/channels/FileChannel/TransferToChannel.java

```diff
@@ -52,52 +71,61 @@ public static void main(String[] args) throws Exception {
         outFile.delete();
     }
 
-    static void test1() throws Exception {
+    private static void test1() throws Exception {
         for (int i=0; i<10; i++) {
             transferFileToUserChannel();
             System.gc();
-            System.err.println("Transferred file...");
+            System.err.println("Transferred file to user channel...");
         }
     }
 
-    static void test2() throws Exception {
+    private static void test2() throws Exception {
         for (int i=0; i<10; i++) {
             transferFileToTrustedChannel();
             System.gc();
-            System.err.println("Transferred file...");
+            System.err.println("Transferred file to trusted channel...");
         }
     }
 
-    static void transferFileToUserChannel() throws Exception {
+    private static void test3() throws Exception {
+        for (int i=0; i<10; i++) {
+            transferFileDirectly();
+            System.gc();
+            System.err.println("Transferred file directly...");
+        }
+    }
+
+    private static void transferFileToUserChannel() throws Exception {
         long remainingBytes = in.size();
         long size = remainingBytes;
         WritableByteChannel wbc = new WritableByteChannel() {
-                Random rand = new Random(0);
-                public int write(ByteBuffer src) throws IOException {
-                    int read = src.remaining();
-                    byte[] incoming = new byte[read];
-                    src.get(incoming);
-                    checkData(incoming, read);
-                    return read == 0 ? -1 : read;
-                }
-                public boolean isOpen() {
-                    return true;
-                }
-                public void close() throws IOException {
-                }
-                void checkData(byte[] incoming, int size) {
-                    byte[] expected = new byte[size];
-                    rand.nextBytes(expected);
-                    for (int i=0; i<size; i++)
-                        if (incoming[i] != expected[i])
-                            throw new RuntimeException("Data corrupted");
-                }
-            };
+            Random rand = new Random(0);
+            public int write(ByteBuffer src) throws IOException {
+                int read = src.remaining();
+                byte[] incoming = new byte[read];
+                src.get(incoming);
+                checkData(incoming, read);
+                return read == 0 ? -1 : read;
+            }
+            public boolean isOpen() {
+                return true;
+            }
+            public void close() throws IOException {
+            }
+            void checkData(byte[] incoming, int size) {
+                byte[] expected = new byte[size];
+                rand.nextBytes(expected);
+                for (int i=0; i<size; i++)
+                    if (incoming[i] != expected[i])
+                        throw new RuntimeException("Data corrupted");
```

Comment:
> Perhaps `if (!Arrays.equals(incoming, 0, size, expected, 0, size))...`

I think here the simpler `Arrays.equals(byte[],byte[])` can be used.

Reply:
> Agree but I would take it further and create a utility method so that it is reusable as it looks like you could take advantage of it later in the test(around line 175)

I think that if line 124 and after uses the simpler `Arrays.equals(byte[],byte[])` then such a utility method would be less useful.

Reply:
Fixed by commit c72b4b58f57d998423fb94b17ac950185109b933.